pkg/health/checker/checker_test.go

```diff
@@ -115,3 +114,75 @@ func TestWatchHealth(t *testing.T) {
 	// blocks _and_ writes to our chan
 	healthChecker.WatchHealth(retChan, errChan, quitChan)
 }
+
+// The behavior of WatchHealth is such that if two values (t0, t1) are emitted
+// via the underlying watch, only t1 will be seen to an interrupted reader
+func TestWatchHealthFreshness(t *testing.T) {
+	fakeKV := kptest.NewFakeKV()
+	healthChecker := &consulHealthChecker{
+		kv: fakeKV,
+	}
+	_ = healthChecker
+
+	retChan := make(chan []*health.Result)
+	errChan := make(chan error)
+	quitChan := make(chan struct{})
+
+	oldStatus := health.HealthState("passing")
+	newStatus := health.HealthState("critical")
+
+	dummyHealthResultOld := &health.Result{
+		ID:      "ID",
+		Node:    "node1.example.com",
+		Service: "Service",
+		Status:  oldStatus,
+		Output:  "output",
+	}
+	dummyBufOld, err := json.Marshal(dummyHealthResultOld)
+	if err != nil {
+		t.Fatalf("json marshal err: %v", err)
+	}
+	_, _, err = fakeKV.CAS(&api.KVPair{Key: "health/service/node1.example.com", Value: dummyBufOld}, nil)
+	if err != nil {
+		t.Fatalf("error: %v", err)
+	}
+
+	dummyHealthResultNew := &health.Result{
+		ID:      "ID",
+		Node:    "node1.example.com",
+		Service: "Service",
+		Status:  newStatus,
+		Output:  "output",
+	}
+	dummyBufNew, err := json.Marshal(dummyHealthResultNew)
+	if err != nil {
+		t.Fatalf("json marshal err: %v", err)
+	}
+	_, _, err = fakeKV.CAS(&api.KVPair{Key: "health/service/node1.example.com", Value: dummyBufNew}, nil)
+	if err != nil {
+		t.Fatalf("error: %v", err)
+	}
+
+	go func() {
+		defer func() {
+			quitChan <- struct{}{}
+		}()
+		for {
```

Comment:
don't think you need this `for` because all cases of the select result in test failure or exit